pkg/apis/v1alpha1/types.go

```diff
@@ -51,6 +51,18 @@ type GatewayClassConfigSpec struct {
 	// +kubebuilder:validation:Enum=trace;debug;info;warning;error
 	// Logging levels
 	LogLevel string `json:"logLevel,omitempty"`
+	// Configuration information about how many instances to deploy
+	DeploymentSpec DeploymentSpec `json:"deployment,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+
+type DeploymentSpec struct {
+	// +kubebuilder:default:=1
+	// +kubebuilder:validation:Maximum=8
+	// +kubebuilder:validation:Minimum=1
+	//Number of gateway instances that should be deployed by default
+	Instances *int32 `json:"defaultInstances,omitempty"`
```

Comment:
I would prefer to have the name here match the yaml config to remove some mental overhead, but that's just preference